### ISSUE TITLE
error NG1010: Value at position 0 in the NgModule.declarations of Module is not a reference

### ISSUE THREAD

**Issue by kumaresan-subramani** [angular/angular#38947]
# 🐞 Bug report

If I do a postinstall ngcc I get the following errors in our library

```
Error: Failed to compile entry-point @syncfusion/ej2-angular-calendars (main as umd) due to compilation errors:
node_modules/@syncfusion/ej2-angular-calendars/dist/ej2-angular-calendars.umd.js:174:31 - error NG1010: Value at position 0 in the NgModule.declarations of CalendarModule is not a reference
  Value could not be determined statically.

174                 declarations: [
                                  ~
175                     exports.CalendarComponent
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
176                 ],
    ~~~~~~~~~~~~~~~~~

  node_modules/@syncfusion/ej2-angular-calendars/dist/ej2-angular-calendars.umd.js:175:21
    175                     exports.CalendarComponent
                            ~~~~~~~~~~~~~~~~~~~~~~~~~
    Unable to evaluate statically.
node_modules/@syncfusion/ej2-angular-calendars/dist/ej2-angular-calendars.umd.js:166:5 - error NG6002: Appears in the NgModule.imports of CalendarAllModule, but could not be resolved to an NgModule class.

Is it missing an @NgModule annotation?

166 var CalendarModule = /** @class */ (function () {
        ~~~~~~~~~~~~~~
node_modules/@syncfusion/ej2-angular-calendars/dist/ej2-angular-calendars.umd.js:166:5 - error NG6003: Appears in the NgModule.exports of CalendarAllModule, but could not be resolved to an NgModule, Component, Directive, or Pipe class.

Is it missing an Angular annotation?
```

### Is this a regression?
Working fine with out `"postinstall": "ngcc"`

## 🔬 Minimal Reproduction

1. download my sample - https://github.com/kumaresan-subramani/Syncfusion-Calendar-Error-1

2. then give `npm i` or `yarn i`

My reported issue will occurs.

## 🔥 Exception or Error

![image](https://user-images.githubusercontent.com/36756050/94011484-0dcaca80-fdc5-11ea-8dac-aa09a9654b51.png)

## 🌍 Your Environment
<pre><code>
    "@angular/animations": "~10.1.0",
    "@angular/common": "~10.1.0",
    "@angular/compiler": "~10.1.0",
    "@angular/core": "~10.1.0",
    "@angular/forms": "~10.1.0",
    "@angular/platform-browser": "~10.1.0",
    "@angular/platform-browser-dynamic": "~10.1.0",
    "@angular/router": "~10.1.0",
    "@angular-devkit/build-angular": "~0.1001.0",
    "@angular/cli": "~10.1.0",
    "@angular/compiler-cli": "~10.1.0",
</code></pre>

**Anything else relevant?**
I have done some researching and found that below issue and tried everything which are reported in those issues but no luck for me.

https://github.com/angular/angular/issues/35255#issuecomment-637055168 - downgraded ts version same error
https://github.com/angular/angular/issues/36673#issuecomment-615140449 - tried changed my [module declaration](https://github.com/syncfusion/ej2-angular-ui-components/blob/master/components/calendars/src/calendar/calendar.module.ts#L8)

### This is our component config - 

https://github.com/syncfusion/ej2-angular-ui-components/tree/master/components/calendars/src/calendar

## Note: i am using `ng-packagr` to generate my angular library

**Comment by petebacondarwin** [comment]
So the reason for this error is that ngcc does not currently support parsing UMD/CommonJS properties of the form `exports.Xxxx`. It doesn't know how to interpret the `exports` identifier.

This is a bug in ngcc. The question is what priority we should give to fixing it.
Our experience is that UMD formats in libraries are usually not required for most use cases. Do you have concrete situations where this format is required?

As you said, if you just let the CLI do the work of running ngcc everything passes. This is because the CLI knows that your project does not need the UMD format, and just gets ngcc to process one of the other formats instead (e.g. ES2015).

By default running `ngcc` with no options will attempt to process every format of every package in node_modules. There are various ways to avoid this, such as adding the `--first-only` property or specifying a list of properties to consider (excluding `main` which maps to UMD) - e.g `-p esm2015 esm5 module browser`. Finally one can exclude the `main` property from just this package by adding a suitably configured `ngcc.config.js` file.

**Comment by kumaresan-subramani** [comment]
@petebacondarwin , i can understand but why `umd` files are generating while using `ng-packagr` to generate angular libraries even `NGC` wont compile this format

**Comment by petebacondarwin** [comment]
Related fix - https://github.com/angular/angular/pull/32424

**Comment by kumaresan-subramani** [comment]
HI @atscott ,

IS this issue fix published in npm package

**Comment by petebacondarwin** [comment]
Hi @kumaresan-subramani - there was a conflict with the patch branch so this PR has not yet been merged there, which means it was not part of the most recent patch release. You can try it out on the `next` dist-tag in npm if you want.
I will create a patch backport PR today, but we have noticed that this fix actually breaks some CommonJS scenarios in our ngcc-validation test repository. So I plan to fix those before landing this PR in patch.

**Comment by kumaresan-subramani** [comment]
@petebacondarwin ,

Thanks for your reply,

Can you please update here once you resolved and provide support

**Comment by petebacondarwin** [comment]
Will do

**Comment by petebacondarwin** [comment]
Here is the fix https://github.com/angular/angular/pull/39267 - to land on master
And here is the back port of the original PR and the one above: https://github.com/angular/angular/pull/39272

**Comment by kumaresan-subramani** [comment]
@petebacondarwin , your mentioned  PR gets closed but without merge.

**Comment by petebacondarwin** [comment]
@kumaresan-subramani - which one? Both PRs mentioned above (#39267 and #39272) have been merged into the appropriate branches. The GH UI doesn't show them as "merged" because our merge script cherry-picks the commits (squashing fixups as necessary) onto the target branch, so GH doesn't realise that the PR was merged.

**Comment by kumaresan-subramani** [comment]
@petebacondarwin , thanks - can i check now with latest npmjs package

**Comment by petebacondarwin** [comment]
I think the fixes appear in 10.1.6
